### PR TITLE
logging: configurable function name prefix

### PR DESCRIPTION
**Pull request by nordic-krch**
Added option to pick which severity levels gets function prefix. Options goes from least severe to the bottom (dbg only **default**, dbg+inf, dbg+inf+wrn, all, none).

@jukkar @jhedberg  I've also modified net and ble log wrappers. I assume they were needed for that missing funcitonality.

### CODE REVIEW

**Comment by jarz-nordic** [comment]
@nordic-krch : Commit message:

> **Exteneded** logger configuration

**Comment by jukkar** [comment]
> I'd rather have "module prefix" optional.

Note that if one wants to process the output file using a grep or similar text processing tool, then it is very useful if the module is there. But of course the module name could be made optional with suitable options if one does not really need them.
Personally I am fine with current logging output, except that I am missing the thread id in debug messages but I had that added in NET_xxx() macros.

**Review by dbkinder — CHANGES_REQUESTED** [review]
wording suggestion change

**Review by jhedberg — CHANGES_REQUESTED** [review]
This PR doesn't seem to make it possible to have the function name in all logs *except* LOG_INF. That's what we've so far had in Bluetooth. The reasoning is as follows:

 - debug: for debugging purposes, function name almost always useful
 - warn/error: implies potential issue in the code, function name helps look up the location
 - info: Information that's intended for the casual user, i.e. someone that's not expected to look into the code itself. An example of such a message is the controller hardware and local address information that the Bluetooth stack logs at startup - it can be useful/interesting for anyone reading the log and having the function name there would add no value (all necessary information is already in the log message itself).

**Comment by nordic-krch** [comment]
@jhedberg @dbkinder ping
Ping to anyone else that could review.

**Review by jhedberg — APPROVED** [review]

**Comment by pfalcon** [comment]
@nordic-krch: This now needs a rebase.

And ping to @dbkinder regarding doc changes.

**Comment by pfalcon** [comment]
@nordic-krch, So one of the issue with this PR is that you change the same lines in same file (the rst doc specifically) in 2 different commits of this PR. Ideally, this should be avoided and the final change should appear just once in one of the commits.

File: include/logging/log.h [hunk 1]
```diff
@@ -21,11 +21,11 @@ extern "C" {
  * @{
  */
 
-#define LOG_LEVEL_NONE  0
-#define LOG_LEVEL_ERR   1
-#define LOG_LEVEL_WRN   2
-#define LOG_LEVEL_INF   3
-#define LOG_LEVEL_DBG   4
+#define LOG_LEVEL_NONE _LOG_LEVEL_NONE
```
Comment (pfalcon) [inline include/logging/log.h]: A curious kid in me really wonders what happens here, but nevermind.

Comment (pfalcon) [reply include/logging/log.h]: I.e. why we can't just define the final non-underscore version in log_core.h and be done with it.

File: subsys/logging/log_output.c [hunk 2]
```diff
@@ -166,8 +167,8 @@ static void color_postfix(struct log_msg *msg,
 }
 
 
-static int ids_print(struct log_msg *msg,
-		     const struct log_output *log_output, bool level_on)
+static int ids_print(struct log_msg *msg, const struct log_output *log_output,
+		     bool level_on, bool func_on)
```
Comment (pfalcon) [inline subsys/logging/log_output.c]: One of next releases, you'll probably want to replace this with a single "flags" param.

**Review by pfalcon — APPROVED** [review]
Looks good enough, but I'd encourage to move all doc changes to the first commit, to avoid polluting git history with interim changes. It wouldn't be too hard here either, as the patchset is small.

File: include/logging/log.h (hunk 1 above)
Comment (nordic-krch) [reply include/logging/log.h]: removed that change. it was not needed.

**Comment by nordic-krch** [comment]
> but I'd encourage to move all doc changes to the first commit, to avoid polluting git history with interim changes. 

Doc changes where accidentally added to the last commit - fixed.

**Review by pfalcon — APPROVED** [review]
Thanks for the updates and continued work on improving the logging, 100% +1 here.

**Review by dbkinder — APPROVED** [review]
docs LGTM

**Comment by nordic-krch** [comment]
@carlescufi @nashif rebased and ready for merge.

**Pull request merged**